app/data_models/list_store.py

```diff
@@ -178,6 +178,12 @@ def _list_item_ids(self) -> list[str]:
 
         return ids
 
+    def get_list_name_for_list_item_id(self, list_item_id: str) -> str:
+        for list_name in self._lists:
+            if list_item_id in self[list_name].items:
+                return list_name
+        raise ValueError(f"list_item_id {list_item_id} not found in any lists")
```

Comment:
If I mess with the url and change id to be invalid, this will result in a 500 error. We could raise 404, but I think we should just return None and in any place where None isn't acceptable, it can raise an exception.

Reply:
Done 👍